sky/jobs/state.py

```diff
@@ -434,7 +442,14 @@ def set_starting(job_id: int, task_id: int, callback_func: CallbackType):
             """\
             UPDATE spot SET status=(?)
             WHERE spot_job_id=(?) AND
-            task_id=(?)""", (ManagedJobStatus.STARTING.value, job_id, task_id))
+            task_id=(?) AND
+            status=(?) AND
+            end_at IS null""", (ManagedJobStatus.STARTING.value, job_id,
+                                task_id, ManagedJobStatus.SUBMITTED.value))
+        if cursor.rowcount != 1:
+            raise exceptions.ManagedJobStatusError(
```

Comment:
Just want to make sure, all those places that raises such error will be correctly caught by the caller, right?

Reply:
These are all called exclusively from the controller logic. It doesn't try to catch `ManagedJobStatusError`, so if this is thrown, it will crash out to FAILED_CONTROLLER. I think this behavior is desirable, since something very unexpected is going on in this case.

Reply:
Yes, crash out to `FAILED_CONTROLLER` is good enough. I just want to make sure it is caught by the path where we set the `FAILED_CONTROLLER`, instead of having it barely thrown (e.g., making sure the place we set the PENDING state does not throw the error, etc)